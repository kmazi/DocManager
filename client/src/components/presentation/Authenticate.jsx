import React from 'react';
import $ from 'jquery';
import PropTypes from 'prop-types';

import SigninForm from './SigninForm';
import SignUpForm from './SignupForm';

 /**
   * shows or hides the signin form
   * @param {object} event - Contains information about the button clicked
   * @return {null} returns void
   */
const showSignInForm = (event) => {
  event.preventDefault();
  const signInForm = $('#signinform');
  const signUpForm = $('#signupform');
  $('#signinbtn').toggleClass('btn homebtncolor');
  $('#signupbtn').toggleClass('btn homebtncolor');
  signUpForm.slideUp(400, () => {
    signInForm.slideDown(400);
  });
};

/**
   * shows or hides the signup form
   * @param {object} event - Contains information about the button clicked
   * @return {null} returns void
   */
const showSignUpForm = (event) => {
  event.preventDefault();
  $('#signupbtn').toggleClass('btn homebtncolor');
  $('#signinbtn').toggleClass('btn homebtncolor');
  const signInForm = $('#signinform');
  const signUpForm = $('#signupform');
  signInForm.slideUp(400, () => {
    signUpForm.slideDown(400);
  });
};

/**
 * Renders the authentication form
 * @return {object} Returns the authentication form to render
 */
const Authenticate = ({ history, signInUser, signUserUp,
  submitButton, roleType, allDocuments }) => (
    <div className="container">
      <div id="authbuttons" className="row">
        <div className="col s6">
          <span
            className="a center-align waves-effect waves-light homebtncolor"
            id="signinbtn"
            href="/"
            onClick={(event) => { showSignInForm(event); }}
          >SignIn</span>
        </div>
        <div className="col s6">
          <span
            className="a center-align waves-effect waves-light btn"
            id="signupbtn"
            href="/"
            onClick={(event) => { showSignUpForm(event); }}
          >SignUp</span>
        </div>
      </div>

      <div id="authform">
        <SigninForm
          history={history}
          signInUser={signInUser}
          signInButtonText={submitButton}
          roleType={roleType}
          allDocuments={allDocuments}
          submitButton={submitButton}
        />

        <SignUpForm
          signUserUp={signUserUp}
          history={history}
          signUpButtonText={submitButton}
          allDocuments={allDocuments}
          roleType={roleType}
          submitButton={submitButton}
        />
      </div>
    </div>
);

Authenticate.propTypes = {
  submitButton: PropTypes.string.isRequired,
  roleType: PropTypes.string.isRequired,
  signUserUp: PropTypes.func.isRequired,
  signInUser: PropTypes.func.isRequired,
  allDocuments: PropTypes.func.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
};

export default Authenticate;
