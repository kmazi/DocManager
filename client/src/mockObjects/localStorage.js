/**
 * Mocks localstorage
 */
class LocalStorageMock {
  /**
   * the constructor
   */
  constructor() {
    this.store = {};
  }
/**
 * clears the localstorage
 * @return {void} returns null
 */
  clear() {
    this.store = {};
  }

/**
 * get item in the localstorage
 * @param {string} key - The key to the item in the localstorage
 * @return {string} returns the key to the item in localstorage
 */
  getItem(key) {
    return this.store[key];
  }
/**
 * set the value of an item in the localstorage class
 * @param {string} key - The key to the item in the localstorage
 * @param {string} value - The value of the item in the localstorage
 * @return {null} returns null
 */
  setItem(key, value) {
    this.store[key] = value;
  }

  /**
 * remove item in the localstorage class
 * @param {string} key - The key to the item in the localstorage
 * @return {void} returns nothing
 */
  removeItem(key) {
    delete this.store[key];
  }
}

global.localStorage = new LocalStorageMock();
export default global.localStorage;
