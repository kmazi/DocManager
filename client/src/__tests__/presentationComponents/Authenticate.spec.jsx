import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Authenticate from '../../components/presentation/Authenticate';
import '../../mockObjects/localStorage';

describe('The Authenticate component:', () => {
  const response = { status: 'successful' };
  const props = {
    history: { push: () => {} },
    signInUser: jest.fn(() => Promise.resolve(response)),
    signUserUp: jest.fn(() => Promise.resolve(response)),
    submitButton: 'submit',
    roleType: 'Learning',
    allDocuments: jest.fn(() => Promise.resolve(response)),
  };
  test('should render correctly', () => {
    const enzymeWrapper = mount(<Authenticate {...props} />);
    const component = enzymeWrapper;
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

  test('should show signup form when signupbtn is clicked',
    () => {
      const event = {
        preventDefault: () => 'Hello',
      };
      const enzymeWrapper = mount(<Authenticate {...props} />);
      enzymeWrapper.find('#authbuttons #signupbtn')
      .props().onClick(event);
      const component = enzymeWrapper;
      const tree = toJson(component);
      expect(tree).toMatchSnapshot();
    });

  test('should show signin form when signinbtn is clicked',
    () => {
      const event = {
        preventDefault: () => 'Hello',
      };
      const enzymeWrapper = mount(<Authenticate {...props} />);
      enzymeWrapper.find('#authbuttons #signinbtn')
      .props().onClick(event);
      const component = enzymeWrapper;
      const tree = toJson(component);
      expect(tree).toMatchSnapshot();
    });
});
