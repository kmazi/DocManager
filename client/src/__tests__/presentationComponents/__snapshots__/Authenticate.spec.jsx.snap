// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Authenticate component: should render correctly 1`] = `
<Authenticate
  allDocuments={[Function]}
  history={
    Object {
      "push": [Function],
    }
  }
  roleType="Learning"
  signInUser={[Function]}
  signUserUp={[Function]}
  submitButton="submit"
>
  <div
    className="container"
  >
    <div
      className="row"
      id="authbuttons"
    >
      <div
        className="col s6"
      >
        <span
          className="a center-align waves-effect waves-light homebtncolor"
          href="/"
          id="signinbtn"
          onClick={[Function]}
        >
          SignIn
        </span>
      </div>
      <div
        className="col s6"
      >
        <span
          className="a center-align waves-effect waves-light btn"
          href="/"
          id="signupbtn"
          onClick={[Function]}
        >
          SignUp
        </span>
      </div>
    </div>
    <div
      id="authform"
    >
      <SigninForm
        allDocuments={[Function]}
        history={
          Object {
            "push": [Function],
          }
        }
        roleType="Learning"
        signInButtonText="submit"
        signInUser={[Function]}
        submitButton="submit"
      >
        <div
          className=""
          id="signinform"
        >
          <input
            className="forminput"
            placeholder="Username"
            type="text"
          />
          <input
            className="forminput"
            placeholder="Password"
            type="password"
          />
          <button
            className="center-align waves-effect waves-light btn"
            id="signin"
            onClick={[Function]}
          >
            submit
          </button>
        </div>
      </SigninForm>
      <SignupForm
        allDocuments={[Function]}
        history={
          Object {
            "push": [Function],
          }
        }
        roleType="Learning"
        signUpButtonText="submit"
        signUserUp={[Function]}
        submitButton="submit"
      >
        <div
          id="signupform"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <input
            className="forminput"
            minLength="3"
            name="username"
            placeholder="Username"
            required={true}
            type="text"
          />
          <input
            className="forminput"
            minLength="10"
            name="email"
            placeholder="Email"
            required={true}
            type="email"
          />
          <input
            className="forminput"
            minLength="6"
            name="password"
            placeholder="Password"
            required={true}
            type="password"
          />
          <input
            className="forminput"
            name="comfirmpassword"
            placeholder="Comfirm password"
            required={true}
            type="password"
          />
          <div
            id="roles"
          >
            Role:  
            <input
              className="with-gap"
              data-value="Learning"
              id="Learning"
              name="group1"
              type="radio"
              value="4"
            />
            <label
              htmlFor="Learning"
            >
              Learning
            </label>
            <input
              className="with-gap"
              data-value="Fellow"
              id="Fellow"
              name="group1"
              type="radio"
              value="3"
            />
            <label
              htmlFor="Fellow"
              id="Fellowlabel"
            >
              Fellow
            </label>
            <input
              className="with-gap"
              data-value="DevOps"
              id="DevOps"
              name="group1"
              type="radio"
              value="5"
            />
            <label
              htmlFor="DevOps"
            >
              DevOps
            </label>
            <br />
          </div>
          <button
            className="center-align waves-effect waves-light btn"
            id="signup"
            onClick={[Function]}
          >
            submit
          </button>
        </div>
      </SignupForm>
    </div>
  </div>
</Authenticate>
`;

exports[`The Authenticate component: should show signin form when signinbtn is clicked 1`] = `
<Authenticate
  allDocuments={[Function]}
  history={
    Object {
      "push": [Function],
    }
  }
  roleType="Learning"
  signInUser={[Function]}
  signUserUp={[Function]}
  submitButton="submit"
>
  <div
    className="container"
  >
    <div
      className="row"
      id="authbuttons"
    >
      <div
        className="col s6"
      >
        <span
          className="a center-align waves-effect waves-light homebtncolor"
          href="/"
          id="signinbtn"
          onClick={[Function]}
        >
          SignIn
        </span>
      </div>
      <div
        className="col s6"
      >
        <span
          className="a center-align waves-effect waves-light btn"
          href="/"
          id="signupbtn"
          onClick={[Function]}
        >
          SignUp
        </span>
      </div>
    </div>
    <div
      id="authform"
    >
      <SigninForm
        allDocuments={[Function]}
        history={
          Object {
            "push": [Function],
          }
        }
        roleType="Learning"
        signInButtonText="submit"
        signInUser={[Function]}
        submitButton="submit"
      >
        <div
          className=""
          id="signinform"
        >
          <input
            className="forminput"
            placeholder="Username"
            type="text"
          />
          <input
            className="forminput"
            placeholder="Password"
            type="password"
          />
          <button
            className="center-align waves-effect waves-light btn"
            id="signin"
            onClick={[Function]}
          >
            submit
          </button>
        </div>
      </SigninForm>
      <SignupForm
        allDocuments={[Function]}
        history={
          Object {
            "push": [Function],
          }
        }
        roleType="Learning"
        signUpButtonText="submit"
        signUserUp={[Function]}
        submitButton="submit"
      >
        <div
          id="signupform"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <input
            className="forminput"
            minLength="3"
            name="username"
            placeholder="Username"
            required={true}
            type="text"
          />
          <input
            className="forminput"
            minLength="10"
            name="email"
            placeholder="Email"
            required={true}
            type="email"
          />
          <input
            className="forminput"
            minLength="6"
            name="password"
            placeholder="Password"
            required={true}
            type="password"
          />
          <input
            className="forminput"
            name="comfirmpassword"
            placeholder="Comfirm password"
            required={true}
            type="password"
          />
          <div
            id="roles"
          >
            Role:  
            <input
              className="with-gap"
              data-value="Learning"
              id="Learning"
              name="group1"
              type="radio"
              value="4"
            />
            <label
              htmlFor="Learning"
            >
              Learning
            </label>
            <input
              className="with-gap"
              data-value="Fellow"
              id="Fellow"
              name="group1"
              type="radio"
              value="3"
            />
            <label
              htmlFor="Fellow"
              id="Fellowlabel"
            >
              Fellow
            </label>
            <input
              className="with-gap"
              data-value="DevOps"
              id="DevOps"
              name="group1"
              type="radio"
              value="5"
            />
            <label
              htmlFor="DevOps"
            >
              DevOps
            </label>
            <br />
          </div>
          <button
            className="center-align waves-effect waves-light btn"
            id="signup"
            onClick={[Function]}
          >
            submit
          </button>
        </div>
      </SignupForm>
    </div>
  </div>
</Authenticate>
`;

exports[`The Authenticate component: should show signup form when signupbtn is clicked 1`] = `
<Authenticate
  allDocuments={[Function]}
  history={
    Object {
      "push": [Function],
    }
  }
  roleType="Learning"
  signInUser={[Function]}
  signUserUp={[Function]}
  submitButton="submit"
>
  <div
    className="container"
  >
    <div
      className="row"
      id="authbuttons"
    >
      <div
        className="col s6"
      >
        <span
          className="a center-align waves-effect waves-light homebtncolor"
          href="/"
          id="signinbtn"
          onClick={[Function]}
        >
          SignIn
        </span>
      </div>
      <div
        className="col s6"
      >
        <span
          className="a center-align waves-effect waves-light btn"
          href="/"
          id="signupbtn"
          onClick={[Function]}
        >
          SignUp
        </span>
      </div>
    </div>
    <div
      id="authform"
    >
      <SigninForm
        allDocuments={[Function]}
        history={
          Object {
            "push": [Function],
          }
        }
        roleType="Learning"
        signInButtonText="submit"
        signInUser={[Function]}
        submitButton="submit"
      >
        <div
          className=""
          id="signinform"
        >
          <input
            className="forminput"
            placeholder="Username"
            type="text"
          />
          <input
            className="forminput"
            placeholder="Password"
            type="password"
          />
          <button
            className="center-align waves-effect waves-light btn"
            id="signin"
            onClick={[Function]}
          >
            submit
          </button>
        </div>
      </SigninForm>
      <SignupForm
        allDocuments={[Function]}
        history={
          Object {
            "push": [Function],
          }
        }
        roleType="Learning"
        signUpButtonText="submit"
        signUserUp={[Function]}
        submitButton="submit"
      >
        <div
          id="signupform"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <input
            className="forminput"
            minLength="3"
            name="username"
            placeholder="Username"
            required={true}
            type="text"
          />
          <input
            className="forminput"
            minLength="10"
            name="email"
            placeholder="Email"
            required={true}
            type="email"
          />
          <input
            className="forminput"
            minLength="6"
            name="password"
            placeholder="Password"
            required={true}
            type="password"
          />
          <input
            className="forminput"
            name="comfirmpassword"
            placeholder="Comfirm password"
            required={true}
            type="password"
          />
          <div
            id="roles"
          >
            Role:  
            <input
              className="with-gap"
              data-value="Learning"
              id="Learning"
              name="group1"
              type="radio"
              value="4"
            />
            <label
              htmlFor="Learning"
            >
              Learning
            </label>
            <input
              className="with-gap"
              data-value="Fellow"
              id="Fellow"
              name="group1"
              type="radio"
              value="3"
            />
            <label
              htmlFor="Fellow"
              id="Fellowlabel"
            >
              Fellow
            </label>
            <input
              className="with-gap"
              data-value="DevOps"
              id="DevOps"
              name="group1"
              type="radio"
              value="5"
            />
            <label
              htmlFor="DevOps"
            >
              DevOps
            </label>
            <br />
          </div>
          <button
            className="center-align waves-effect waves-light btn"
            id="signup"
            onClick={[Function]}
          >
            submit
          </button>
        </div>
      </SignupForm>
    </div>
  </div>
</Authenticate>
`;
